ckan/views/home.py

```diff
@@ -0,0 +1,85 @@
+# encoding: utf-8
+
+from flask import Blueprint, render_template, abort
+from flask.views import View
+
+import ckan.model as model
+import ckan.logic as logic
+import ckan.lib.search as search
+from ckan.common import g, c, config, _
+import ckan.lib.helpers as h
+
+CACHE_PARAMETERS = [u'__cache', u'__no_cache__']
+
+home = Blueprint(u'home', __name__)
+
+
+@home.before_request
+def before_request():
+    try:
+        context = {
+            u'model': model,
+            u'user': g.user,
+            u'auth_user_obj': g.userobj
+        }
+        # views.identify_user()
+        logic.check_access(u'site_read', context)
+    except logic.NotAuthorized:
+        abort(403)
+
+
+class HomeView(View):
+    def dispatch_request(self):
+        try:
+            # package search
+            context = {u'model': model, u'session': model.Session,
+                       u'user': g.user, u'auth_user_obj': g.userobj}
+            data_dict = {
+                u'q': u'*:*',
+                u'facet.field': h.facets(),
+                u'rows': 4,
+                u'start': 0,
+                u'sort': u'views_recent desc',
+                u'fq': u'capacity:"public"'
+            }
+            query = logic.get_action(u'package_search')(
+                context, data_dict)
+            g.search_facets = query['search_facets']
+            g.package_count = query['count']
+            g.datasets = query['results']
```

Comment:
As mentioned in https://github.com/ckan/ckan/pull/3203 we should take the opportunity of the rewrite for deprecating as much of these as possible (ie we still keep them in `g` for a version, but we pass explicitly the vars to the templates). This might not be possible with all of them (like in `g.search_facets_limits`) but if possible, let's deprecate them.